packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -526,23 +526,38 @@ export class DatabaseCluster extends DatabaseClusterNew {
    * Adds the single user rotation of the master password to this cluster.
    */
   public addRotationSingleUser(options: RotationSingleUserOptions = {}): secretsmanager.SecretRotation {
+    if (options.secret) {
+      //TODO: check given secret is not master secret of this!
+      //TODO: check given secret is not added already!
+      //TODO: how to generate proper ID?
+      const id = 'AdditionalSecretRotation';
+      return this.newSecretRotation(id, options);
+    } else {
+      return this.addRotationSingleUserMasterSecret(options);
+    }
+  }
+
+  private addRotationSingleUserMasterSecret(options: RotationSingleUserOptions): secretsmanager.SecretRotation {
     if (!this.secret) {
-      throw new Error('Cannot add single user rotation for a cluster without secret.');
+      throw new Error('Cannot add single user rotation for a cluster without master secret.');
```

Comment:
```suggestion
      throw new Error('Cannot add single user master secret rotation for a cluster without a master secret.');
```

Reply:
Rotation is for a secret, so the first "master secret" seems redundant. Let me know if you still want it there.